radis/reports/api/viewsets.py

```
@@ -264,185 +371,246 @@ def on_commit():
+        response_data = await sync_to_async(
+            lambda: serializer.data, thread_sensitive=True
+        )()
+        return Response(response_data, status=status.HTTP_201_CREATED)
+
+    async def aretrieve(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        try:
+            report = await Report.objects.select_related("language").aget(
+                document_id=kwargs[self.lookup_field]
+            )
+        except Report.DoesNotExist:
+            raise Http404
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)